src/d3a/d3a_core/sim_results/stats.py

```diff
@@ -245,17 +246,25 @@ def _generate_external_and_total_bills(self, area, results, flattened):
             # External trades are the trades of the parent area
             external = flattened[area.name].copy()
             # Should not include market fee to the external trades
-            market_fee = external.pop("market_fee", None)
+            market_fee = external.pop("market_fee", 0.)
             # Should switch spent/earned and bought/sold, to match the perspective of the UI
-            spent = external.pop("spent")
+            spent = external.pop("spent") + market_fee
             earned = external.pop("earned")
             bought = external.pop("bought")
             sold = external.pop("sold")
             external.update(**{"spent": earned, "earned": spent, "bought": sold,
-                               "sold": bought, "market_fee": market_fee})
+                               "sold": bought, "market_fee": 0.})
             external["total_energy"] = external["bought"] - external["sold"]
             external["total_cost"] = external["spent"] - external["earned"]
             results[area.name].update({"External Trades": external})
+            results[area.name]["Accumulated Trades"]["bought"] += external["bought"]
+            results[area.name]["Accumulated Trades"]["sold"] += external["sold"]
+            results[area.name]["Accumulated Trades"]["spent"] += external["spent"]
+            results[area.name]["Accumulated Trades"]["earned"] += \
```

Comment:
As a followup to my other comment, the Accumulated Trades grid fees need to also be a sum of the grid fees of its children (including External Trades). Reason I am not adding it there, is because this code does not appear in the PR and cannot leave a comment. I am referring to line 242 of this file.